### ISSUE TITLE
NPCs with the staying personality can land on wormholes and leave the system

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#5943]
**Describe the bug**
An NPC with the staying personality that is in a system with a landable wormhole may randomly choose to land on the wormhole, causing it to leave the system. This breaks the staying personality, then causing the NPC to fly between systems randomly as if the staying personality wasn't there.

**To Reproduce**
1. Spawn an NPC with the staying personality in a system with a wormhole that it can land on. (Keystone Research 5 and 6 both create this possibility, and this is where this bug is being observed. I'll note though that this NPC also has the surveillance personality, which changes its idle behavior. Not sure if that changes anything.)
2. Wait as the NPC idles around.
3. Observe how the NPC will eventually choose the wormhole and land on it.

**Expected behavior**
Ships with the staying personality should not leave the system under any circumstances, including via wormhole.

**Screenshots**
N/A

**System (please complete the following information):**
 - OS: Windows 10
 - Game Source: Self-compiled
 - Version: confirmed this bug back in Janurary around commit https://github.com/endless-sky/endless-sky/commit/ff201262f23202d10e5672e2f4a16a93692d73f7. Don't have time to reconfirm it but I have seen at least one report since v0.9.13 came out of this occurring.

**Link to save file**
None.

**Additional context**
N/A

**Comment by MasterOfGrey** [comment]
Is the solution to this not usually to have them use a government that won't use the wormhole?

**Comment by beccabunny** [comment]
> Is the solution to this not usually to have them use a government that won't use the wormhole?

That's not a solution, it's just going around the bug

**Comment by jostephd** [comment]
Here's a test case:

<details>

<summary>
test case (add the Hai wormhole to Rutilicus)
</summary>

```diff
diff --git a/data/intro missions.txt b/data/intro missions.txt
index f0fabfd84..f4b1b503a 100644
--- a/data/intro missions.txt	
+++ b/data/intro missions.txt	
@@ -487,3 +487,29 @@ conversation "end of intro missions"
 		
 	label end
 	`	He hands you a credit chip worth <payment>, and says, "Best of luck to you, Captain. Don't hesitate to look me up if you're ever in the area. And remember to check the spaceports wherever you land. You never know who you'll find wandering around there."`
+
+mission "test"
+	name "test 5963"
+	description "test"
+	repeat
+	landing
+	on offer
+		conversation
+			`	Spawn a staying ship?`
+			choice
+				`Yes, and launch.`
+					launch
+				`Yes.`
+					accept
+				`No.`
+					decline
+	npc kill
+		personality timid appeasing staying waiting uninterested target marked
+		government Merchant
+		fleet
+			names "republic capital"
+			fighters "republic fighter"
+			variant
+				"Carrier (Mark II)"
+				"Lance" 4
+				"Combat Drone" 6
diff --git a/data/map.txt b/data/map.txt
index 65fc56508..341bf0b5d 100644
--- a/data/map.txt
+++ b/data/map.txt
@@ -20827,6 +20827,10 @@ system Rutilicus
 		sprite planet/rock6
 		distance 158.61
 		period 31.9607
+	object "Wormhole Alpha"
+		sprite planet/wormhole
+		distance 208.63
+		period 2158.5
 	object "New Boston"
 		sprite planet/cloud6
 		distance 513.86
@@ -24440,10 +24444,6 @@ system "Ultima Thule"
 			sprite planet/ice8-b
 			distance 216
 			period 11.1964
-	object "Wormhole Alpha"
-		sprite planet/wormhole
-		distance 4208.63
-		period 2158.5
 
 system Umbral
 	pos -164 406
```

</details>

With this, I started a new pilot, waved James off, and accepted the test mission. The Merchant Carrier flew around, including to the wormhole, but didn't land on it.

I then added the `surveillance` personality. That caused the Carrier to launch its fighters/drones and jump out of the system, despite the `staying` personality.

**Comment by warp-core** [comment]
Fixed by #7499.
